action/flow/event/flowevents.go

```diff
@@ -0,0 +1,76 @@
+package event
+
+import (
+	"time"
+)
+
+type Status string
+
+const (
+	CREATED   = "Created"
+	COMPLETED = "Completed"
+	CANCELLED = "Cancelled"
+	FAILED    = "Failed"
+	SCHEDULED = "Scheduled"
+	SKIPPED   = "Skipped"
+	STARTED   = "Started"
+	WAITING   = "Waiting"
+	UNKNOWN   = "Created"
+)
+
+const FLOW_EVENT_TYPE = "flowevent"
+const TASK_EVENT_TYPE = "taskevent"
+
+
+// FlowEvent provides access to flow instance execution details
+type FlowEvent interface {
+	// Returns flow name
+	Name() string
+	// Returns flow ID
+	ID() string
+	// In case of subflow, returns parent flow name
+	ParentName() string
+	// In case of subflow, returns parent flow ID
+	ParentID() string
+	// Returns event time
+	Time() time.Time
+	// Returns application name
+	AppName() string
+	// Returns application version
+	AppVersion() string
+	// Returns current flow status
+	Status() Status
+	// Returns output data for completed flow instance
+	Output() map[string]interface{}
+	// Returns error for failed flow instance
+	Error() error
+}
+
+
+// TaskEvent provides access to task instance execution details
+type TaskEvent interface {
+
+	// Returns flow name
+	FlowName() string
+	// Returns flow ID
+	FlowID() string
+	// Returns task name
+	Name() string
+	// Returns task type
+	Type() string
+	// Returns task status
+	Status() Status
+	// Returns application name
+	AppName() string
+	// Returns application version
+	AppVersion() string
+	// Returns event time
+	Time() time.Time
+	// Returns task input data
+	TaskInput() map[string]interface{}
```

Comment:
I noticed that you have Task in front of Input & Output, but not in other places.  In the FlowEvent you have just Output.  I think we should be consistent.  Maybe add Task in front of everything that comes directly from task?  A little redundant.. but right now if you leave out task Time() isn't very clear... is it the event time or task time (whatever that is)

Reply:
@fm-tibco : How about this?
```
type FlowEvent interface {
	// Returns flow name
	FlowName() string
	// Returns flow ID
	FlowID() string
	// In case of subflow, returns parent flow name
	ParentFlowName() string
	// In case of subflow, returns parent flow ID
	ParentFlowID() string
	// Returns event time
	EventTime() time.Time
	// Returns application name
	AppName() string
	// Returns application version
	AppVersion() string
	// Returns current flow status
	FlowStatus() Status
	// Returns output data for completed flow instance
	FlowOutput() map[string]interface{}
	// Returns error for failed flow instance
	FlowError() error
}


// TaskEvent provides access to task instance execution details
type TaskEvent interface {

	// Returns flow name
	FlowName() string
	// Returns flow ID
	FlowID() string
	// Returns task name
	TaskName() string
	// Returns task type
	TaskType() string
	// Returns task status
	TaskStatus() Status
	// Returns application name
	AppName() string
	// Returns application version
	AppVersion() string
	// Returns event time
	EventTime() time.Time
	// Returns task input data
	TaskInput() map[string]interface{}
	// Returns task output data for completed task
	TaskOutput() map[string]interface{}
	// Returns error for failed task
	TaskError() error
}
```

Reply:
@fm-tibco : PR is updated